compiler/modules/nback/number.bal

```diff
@@ -138,8 +137,11 @@ function buildConvertToInt(llvm:Builder builder, Scaffold scaffold, bir:ConvertT
     buildConvertFloatToInt(builder, scaffold, buildUntagFloat(builder, scaffold, tagged), insn);
     builder.br(joinBlock);
 
-    builder.positionAtEnd(<llvm:BasicBlock>noFloatBlock);
-    if !t:isSubtypeSimple(semType, t:FLOAT) {
+    builder.positionAtEnd(noFloatBlock);
+    if t:isSubtypeSimple(insn.result.semType, t:INT) {
```

Comment:
This doesn't look good to me.  We should be looking at the Repr to decide whether to convert the representation not the SemType.  There are a bunch of functions already that deal with converting between tagged/untagged representations.  If there aren't ones that work for this case, we should add them.

Reply:
But this is going to be too complicated to inline especially with decimal. Better to rewrite using a runtime function.

Reply:
May I add a rt function in this PR?